### ISSUE TITLE
Doubt about YOLOv3 loss

### ISSUE THREAD

**Issue by heavengate** [pjreddie/darknet#1334]
I have some doubt about YOLOv3 loss function. Could someone help me?
1. In paper, binary cross entropy loss is used for objectness loss and classification loss, but in darknet implement, all losses use square loss, so which one should I use.
2. In paper, bounding box prior which is not the best but does overlap a ground truth object by more than 0.5 is ignored, this seems didn't implement in darknet code.

Thanks!

**Comment by rishucoding** [comment]
Hello @heavengate , can you share the paper link?

**Comment by heavengate** [comment]
Paper link is in https://pjreddie.com/media/files/papers/YOLOv3.pdf 
code is in https://github.com/pjreddie/darknet/blob/master/src/yolo_layer.c
@rishucoding

**Comment by jingtianyilong** [comment]
Same question here.
If you search for cost_layer. you will find following type: 
SSE; MASKED; SMOOTH; L1;
And my clue stops here. I can't find anything that relates to binary cross entropy.

**Comment by shivin9** [comment]
Any update? I am doing a classification task and my model is not learning properly. Is this the cause?

**Comment by lz20061213** [comment]
In yolo_layer.c, the gradient is alread calcuated and saved to l.delta during forward. 
And I think the l.cost is not the true loss, it's just used for showing.

**Comment by AlexeyAB** [comment]
> And I think the l.cost is not the true loss, it's just used for showing.

Yes.
